File: test-suite/src/test/java/io/micronaut/docs/config/env/OrderTest.java
package io.micronaut.docs.config.env;

import io.micronaut.context.ApplicationContext;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

class OrderTest {

    @Test
    void testOrderOnFactories() {
        ApplicationContext applicationContext = ApplicationContext.run();
        List<RateLimit> rateLimits = applicationContext.streamOfType(RateLimit.class)
                .toList();

        assertEquals(
                2,
                rateLimits.size()
        );
        assertEquals(1000L, rateLimits.get(0).getLimit().longValue());
        assertEquals(100L, rateLimits.get(1).getLimit().longValue());

        applicationContext.close();
    }
}

File: test-suite-kotlin2-ksp2/src/test/kotlin/io/micronaut/docs/config/env/OrderTest.kt
package io.micronaut.docs.config.env

import io.micronaut.context.ApplicationContext
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import java.util.stream.Collectors

class OrderTest {

    @Test
    fun testOrderOnFactories() {
        val applicationContext = ApplicationContext.run()
        val rateLimits = applicationContext.streamOfType(RateLimit::class.java)
                .collect(Collectors.toList())
        assertEquals(
                2,
                rateLimits.size
                        .toLong())
        assertEquals(1000L, rateLimits[0].limit.toLong())
        assertEquals(100L, rateLimits[1].limit.toLong())
        applicationContext.close()
    }
}
